pkg/gpu/cuda/fatbin.go

```diff
@@ -52,6 +52,16 @@ func (fb *Fatbin) GetKernel(name string, smVersion uint32) *CubinKernel {
 	return fb.Kernels[key]
 }
 
+// HasKernelWithName returns true if the fatbin has a kernel with the given name
+func (fb *Fatbin) HasKernelWithName(name string) bool {
+	for key := range fb.Kernels {
+		if key.Name == name {
```

Comment:
Can you create a `map[string]struct{}` ahead of time for the kernel names? That way you don't have to do this loop for each symbol.

Reply:
Done!